CefSharp.BrowserSubprocess.Core/CefAppUnmanagedWrapper.cpp

```diff
@@ -91,64 +91,105 @@ namespace CefSharp
     {
         auto handled = false;
         auto name = message->GetName();
-        if (name == kEvaluateJavascriptRequest)
+        auto argList = message->GetArgumentList();
+
+        auto browserId = argList->GetInt(0);
+        if (browser->GetIdentifier() != browserId)
         {
-            auto argList = message->GetArgumentList();
-            auto browserId = argList->GetInt(0);
-            auto frameId = GetInt64(argList, 1);
-            auto callbackId = GetInt64(argList, 2);
-            auto script = argList->GetString(3);
+            throw gcnew InvalidOperationException(String::Format("Request BrowserId : {0} does not match browser Id : {1}", browserId, browser->GetIdentifier()));
```

Comment:
So I made a minor change which makes things a little more explicit. I think some of it still needs a little refactoring.

https://github.com/amaitland/CefSharp/commit/d766cc826e8caecc025a3db8d48d2bd49c3c305f#diff-2731f23ba4e8fcf96f8501a49c6d687eL865

Basically the call to `SendProcessMessage` should happen on the browser instance that it was intended for, currently it's executing on the main browser process. When can then remove the `browserId` param.

https://github.com/amaitland/CefSharp/commit/d766cc826e8caecc025a3db8d48d2bd49c3c305f#diff-a27ca20dcc4f76e2dde0c785dd2abb9cR104
For the the `kJavascriptCallbackDestroyRequest` message, if it can't find the browser wrapper then It's pretty much safe to ignore as it's pretty much guaranteed to have been disposed already (unless there's a memory leak somewhere else).

Some of it's still a little rough, so I think it's time to revisit the whole structure. I've been thinking how we can move a large portion of the code into `c#` land with a few simple wrappers, so I'll toy with that idea a little more.